src/extreme_filter.jl

```diff
@@ -268,3 +267,206 @@ Base.@propagate_inbounds function _minimum_fast(A::AbstractArray{Bool}, p, Ω)
     end
     return rst
 end
+
+function _padded_copyto!(dest::AbstractVector, src::AbstractVector, shift, v)
+    Base.require_one_based_indexing(dest, src)
+    if shift >= 0
+        o = shift
+        vsrc = @view src[(1+o):end]
+        N = length(vsrc)
+        copyto!(dest, 1, vsrc, 1, N)
+        vdest = @view dest[(N+1):end]
+        fill!(vdest, v)
+    else
+        o = -shift
+        vdest = @view dest[1:o]
+        fill!(vdest, v)
+        copyto!(dest, 1 + o, src, 1, length(dest) - o)
+    end
+    return dest
+end
+
+# ptr level optimized implementation for Real types
+# short-circuit all check
+function _unsafe_padded_copyto!(dest::AbstractVector, src::AbstractVector, dir, N, v) where {T}
```

Comment:
I'm pretty sure we need to restrict the type annotation to `Vector` (array types with valid pointer for memcopy) for safety.

Reply:
I think you're right, do you want if fix it or could you do this on your side

Reply:
When you're satisfied with the implementation, I can do the cleanup work before merging.

Here's my understanding of the todo work:

- further tweak the performance (trying my previous comments and maybe more)
- call the optimized implementation from `extreme_filter!` under certain condition
- maybe a few more tests and benchmarks

Let me know if you plan to do all of them or just the first performance tweak.

Reply:
I could try to do points 1,3,4 (i will made a short try to avoid unecessay copy, but it could be in another ieteration), i will add comment and docs also  
for the point 2 you are more qualified than me to ensure all safety conditions are taking into account
I try to do this today!
And thanks again of course